consensus/src/state_computer.rs

```diff
@@ -44,15 +49,20 @@ impl ExecutionProxy {
     ) -> StateComputeResult {
         let execution_block_response = execution_proto::ExecuteBlockResponse::from_proto(response)
             .expect("Couldn't decode ExecutionBlockResponse from protobuf");
-        let execution_duration_ms = pre_execution_instant.elapsed().as_millis();
+        let execution_duration = pre_execution_instant.elapsed();
         let num_txns = execution_block_response.status().len();
         if num_txns == 0 {
             // no txns in that block
-            counters::EMPTY_BLOCK_EXECUTION_DURATION_MS.observe(execution_duration_ms as f64);
+            counters::EMPTY_BLOCK_EXECUTION_DURATION_S.observe_duration(execution_duration);
         } else {
-            counters::BLOCK_EXECUTION_DURATION_MS.observe(execution_duration_ms as f64);
-            let per_txn_duration = (execution_duration_ms as f64) / (num_txns as f64);
-            counters::TXN_EXECUTION_DURATION_MS.observe(per_txn_duration);
+            counters::BLOCK_EXECUTION_DURATION_S.observe_duration(execution_duration);
+            if let Ok(nanos_per_txn) =
+                u64::try_from(execution_duration.as_nanos() / num_txns as u128)
+            {
+                // TODO: use duration_float once it's in stable
```

Comment:
nit: you may want to link to https://github.com/rust-lang/rust/issues/54361 in comments